plugins/AudioFileProcessor/AudioFileProcessorWaveView.cpp

```diff
@@ -480,7 +480,8 @@ void AudioFileProcessorWaveView::reverse()
 
 void AudioFileProcessorWaveView::updateCursor(QMouseEvent * me)
 {
-	const auto pos = position(me);
+	QPoint pos;
+	if (me) { pos = position(me); }
```

Comment:
I have the same remark as @Rossmaxx had before. The current code is confusing because on first sight it looks as if `QPoint(0, 0)` could be used in a sensible way if `me == nullptr`. To get rid of the initialization above I propose to extract a method that might look as follows:
```
bool AudioFileProcessorWaveView::closeToStartEndOrLoop(QMouseEvent * me)
{
    if (me == nullptr)
    {
        return false;
    }

    const QPoint pos = position(me);

    return isCloseTo(pos.x(), m_startFrameX) || isCloseTo(pos.x(), m_endFrameX) || isCloseTo(pos.x(), m_loopFrameX);
}
```
This method would then simply be called to set `pointerCloseToStartEndOrLoop`:
```
bool const pointerCloseToStartEndOrLoop = closeToStartEndOrLoop(me);
```

This also gives a better separation of concerns and makes the code more readable.

Reply:
huh, good idea actually. I hope the liberties I took with the exact impl of that are fine.